crates/editor/src/editor.rs

```diff
@@ -1179,6 +1183,10 @@ impl Editor {
                 project_subscriptions.push(cx.subscribe(project, |editor, _, event, cx| {
                     if let project::Event::RefreshInlayHints = event {
                         editor.refresh_inlay_hints(InlayHintRefreshReason::RefreshRequested, cx);
+                    } else if let project::Event::LanguageServerAdded(_, _, _)
+                    | project::Event::LanguageServerRemoved(_) = event
+                    {
+                        editor.refresh_diagnostics(cx);
```

Comment:
There is something I don't understand currently. Subscribing here to these events break the `test_move_to_enclosing_bracket` test. Surely I do know that some of my changes caused that but I just have no idea. Will continue my investigation.

Reply:
Removing the debounce in `Editor::refresh_diagnostics makes tests green:

```
cx.background_executor()
  .timer(DOCUMENT_DIAGNOSTICS_DEBOUNCE_TIMEOUT)
  .await;
```

I guess I am blocking something there when I start pulling diagnostics upoon receiving `project::Event::LanguageServerAdded` or `LanguageServerRemoved`.

Reply:
Simply waiting for the debounce makes the test green. Tho, I am not sure this is the good fix.

Reply:
Yeah, this is very odd and somewhat concerning.
This fix could've been fine if the test failed on something like "something is not parked and we're blocked on waiting for LSP response" but as I see when running this test from the branch without the debounce, that we simply not move the caret.
So, the fix is not good and we need to dig deeper to understand why does not it move.

Reply:
Just chatted about that as it surfaced on another branch: seems that we can use `cx.run_until_parked()` instead, as 

> the test replaces an editor's text but does not necessarily wait for the parsing to complete
> and then tries to use the syntax tree

So, not an issue of the current PR.
This will get fixed in another branch and gets merged, but we can change it here too, just to be on par with a proper fix.

Reply:
Thanks! `cx.run_until_parked()` worked!